### PR TITLE
Scaffolds within specified folders supporting Unity scaffold

### LINKED ISSUE #888: Support specfying scaffolds inside folders
Add the ability to provide a folder location within a repo that should only be used for the scaffold. 

This will be used to enable adding Unity scaffolds since they are all inside the samples folder.

### PR DESCRIPTION
**Pull request by sideninja**
Closes #888 

## Description
Support specfying folder using scaffolds, so the same repo can contain multiple scaffolds inside each folder. This will unlock adding unity scaffolds. 

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: scaffolds.json [hunk 1]
```diff
@@ -16,5 +16,19 @@
...
+    "name": "Simple Unity",
+    "repo": "https://github.com/onflow/UnityFlowSDK.git",
+    "description": "Simple examples how to interact with Flow network using Unity SDK.",
+    "commit": "d01a4b14a98383c18b4d57e87624cb8dba6d0946",
+    "folder": "Samples~/FlowControlQuickstartCodeSample"
+  },
+  {
+    "name": "Mobile Unity Game",
+    "repo": "https://github.com/onflow/UnityFlowSDK.git",
+    "description": "Example Unity words game built on Flow using the Unity SDK.",
```
Comment (DylanTinianov) [inline scaffolds.json:30]: Don't need to say Unity twice, could just be: `"Example word game built on Flow using the Unity SDK."`

File: scaffolds.json [hunk 2]
```diff
@@ -16,5 +16,19 @@
     "repo": "https://github.com/onflow/scaffold-flow.git",
     "description": "Simple demo application using next.js and FCL with provided Cadence contracts.",
     "commit": "4158c5e11b6d2c9e1cbc99af597e3b42b2e6c193"
+  },
+  {
+    "name": "Simple Unity",
+    "repo": "https://github.com/onflow/UnityFlowSDK.git",
+    "description": "Simple examples how to interact with Flow network using Unity SDK.",
```
Comment (DylanTinianov) [inline scaffolds.json:23]: Could be: `"Simple example demonstrating how to interact with the Flow network using Unity SDK."`

**Review by DylanTinianov — APPROVED** [review]
Small comments on descriptions, but looks good.

File: scaffolds.json (hunk 1 above)
Comment (sideninja) [reply scaffolds.json:30]: thank you! this is better for sure

Comment (sideninja) [reply scaffolds.json]: addressed

**Pull request merged**